ocaml/fstar-lib/FStarC_Parser_LexFStar.ml

```diff
@@ -473,6 +473,7 @@ match%sedlex lexbuf with
  | "#print-effects-graph" -> PRAGMA_PRINT_EFFECTS_GRAPH
  | "__SOURCE_FILE__" -> STRING (L.source_file lexbuf)
  | "__LINE__" -> INT (string_of_int (L.current_line lexbuf), false)
+ | "__FL__"   -> STRING ((L.source_file lexbuf) ^ "(" ^ (string_of_int (L.current_line lexbuf)) ^ ")")
```

Comment:
`__FL__` seems a bit obscure to me, should we maybe call this `__FILELINE__`? But I don't have a strong opinion, would be good if others chime in.

Reply:
You have to reference it in every test so long is kind of cumbersome. For example: 
let test_system n =
 let r = FStar.Unix.system "echo HI" in
   match r with 
   | WEXITED e -> if_test    __FL__ (e = 0) n None
   | _         -> final_fail __FL__         n None

Reply:
I agree with Guido, I aslo find FL a bit obscure.
IMO, aggregating __LINE__ and __SOURCE_FILE__ in this formatting is quite opinionated and should probably be user-defined.

@briangmilnes why not using a tactic for that here, e.g.:
```fstar
module Hello
open FStar.Char
open FStar.Tactics

let fl
  (x: unit)
  (#[let sealed_range = range_of_term (quote x) in
     let range = unseal sealed_range in
     let result = FStar.Range.explode range in
     exact (quote result)] loc: string * int * int * int * int)
 = let file, line_start, col_start, _line_end, _col_end = loc in
   file ^ "(" ^ string_of_int line_start ^ ")"

// This is line 14
let file_line: string = fl ()
let _ = assert_norm (file_line == "<input>(15)")
```

Reply:
This is for logging and test packages. Nothing should be left to the user. In testing the ocaml wrap of unix and a few other things, I need the file/line 139 times. Terse is good here, although I eschew it in almost all situations.

A tactic that allows a function is a great alternative. The language server in emacs takes that and works but I can't get a command line to compile it. What's the change needed for compilation? It also has the advantage that the strings are not coming back from the lexer with strange UTF8.

Reply:
Well, verbosity or terseness is a choice, right? I tend to prefer verbosity in such cases, but that's only my preference.

Nice! Here it won't work via the command line because of `<input>`: that's the file name given by the F* interactive protocol used by emacs I believe.
Running that from a file whose path is `/tmp/Hello.fst`, you should be able to `assert_norm (file_line == "/tmp/Hello.fst(15)")` (instead of the one I've put above).

Reply:
The issue I have is it will not compile, not the string back from fl() or __FL__. 

F*orge: VALIDATE _build/fstar/fst/checked/Hello.fst.checked
* Error 168 at src/Hello.fst(8,8-8,8):
  - Syntax error

1 error was reported (see above)
That's the (# line on my test of it running in OCaml.